File: Class/1.rb
class Cal #루비는 무조건 대문자로 시작
    def initialize(v1,v2) #initialize는 생성자를 만드는 작업을 함 
        # @를 통해서 매개변수 값을 받아줘야함. @v1(인스턴스 변수)
        @v1 = v1 # @가 없는 v1는 지역변수임
        @v2 = v2
    end

    def add()
        return @v1+@v2
    end
    
    def subtract()
        return @v1-@v2
    end
end

c1 = Cal.new(10,10)
p(c1.add()) #p 또한 puts와 같은 출력 함수임
p(c1.subtract())

c2 = Cal.new(30,20)
p(c2.add())
p(c2.subtract())

File: Class/1.py
class Cal(object) :
    def __init__(self,v1,v2) : #생성자 self는 생성한 인스턴스가 들어옴
        self.v1 = v1
        self.v2 = v2
    def add(self) : # self 첫 번째에는 생성 인스턴스가 오고 이름은 마음대로 바꿔도 된다.
        return self.v1+self.v2
    def subtract(self) :
        return self.v1 - self.v2

c1 = Cal(10,10) #첫 번째 매개변수는 2번째, 두 번째 매개변수는 첫번째를 가리킴
print(c1.add())
print(c1.subtract())

c2 = Cal(30,20)
print(c2.add())
print(c2.subtract())